### ISSUE TITLE
How to use mask to train loacal area of ourselves trained image data

### ISSUE THREAD

**Issue by ltshan** [nianticlabs/monodepth2#223]
Hi, 

   Thansk for your great work firstly. 
   We already found there exist some occlusion in our trained stereo image data ,so we decide to use mask to avoid these local occluded area joining network training. 
   My question is how to enable this mask funciton in this project.  setting zero value to source rectified images directly or ensure mask area not join loss computation or other better ways?
   pls help provide some instructions?

thanks

**Comment by daniyar-niantic** [comment]
Hi @ltshan 

If you have masks that specify where pixels are occluded, I think the best strategy would be to apply those masks before computing the reprojection loss: https://github.com/nianticlabs/monodepth2/blob/master/trainer.py#L484

So, if you have a `mask`, then something like this should work:
```
loss += to_optimise[mask].mean()
```
